### DISCUSSION TITLE
Interpreting power_test output

### DISCUSSION THREAD

**Discussion by gjones1219** [jazznbass/scan#67] [Q&A] [answered]
Hello Jürgen,

Thank you so much for creating this package! I am starting to use it for some single case analyses I am embarking upon and its very helpful.

I had a question about interpreting output from the power_test function, as it was not clear to me how to interpret some of it. Specifically, what should we make of the p-value column in the power_test function. More specifically, when we have a power_test output with a high alpha error, but a p-value that indicates a significant test, what should we make of this output.

I attach an example below

set.seed(1)
###to fix randomness in place

design <- design(
  n = 1,
  phase_design = list(A = 3, B = 8),
  level = list(A = 0, B = .3),
  trend = 0.02,
  distribution = "normal",
 
)
###including a design with a small effect size for phase b

res <- power_test(design,  n_sim = 100)
res

###When looking at the output, one sees that the randomness test has the most power and also it's p-value is below 0.05. But, I am not sure whether this analysis is the one to go with, as its alpha error is 35 (in the scanBook, you previously referred to an alpha level of 25 as "unacceptably high"). Thus, not sure which metrics I should use to ultimately evaluate which analytical choice is best. Guidance here would be appreciated, thank you!

Grant

**Comment by jazznbass** [comment, marked as answer]
The p-value is a simple binomial test of the proportion "correct" against 50%. It does not tell you much as it depends on the number of simulations. I mainly use it to get an idea if the number of simulations is high enough. But it is only a rough estimate.

I increase n_sim to 10000:

```.r
res <- power_test(design, n_sim = 10000)
res
Test-Power in percent:

    Method Power Alpha Error Alpha:Beta Correct     p
 plm_level  7.32        5.09     1:18.2    51.1 0.002
      rand 59.16       39.91      1:1.0    59.6 0.000
      tauU 14.24        4.94     1:17.4    54.6 0.000
```

Basically, you do not have enough power to detect such a small effect with this few measurements.

If possible, go for a design with more measurements and a higher minimal effect you can detect:

```.r
design <- design(
  n = 3,
  phase_design = list(A = 5, B = 10),
  level = list(A = 0, B = 0.8),
  trend = 0.02,
  distribution = "normal", 
)
```

yields:

```.r
res <- power_test(design, method = c("hplm_level", "tauU_meta", "rand"), n_sim = 500)
res

Test-Power in percent:

     Method Power Alpha Error Alpha:Beta Correct p
 hplm_level  60.6         5.2      1:7.6    77.7 0
  tauU_meta 100.0        12.4      1:0.0    93.8 0
       rand  86.6        10.6      1:1.3    88.0 0
```
In this example I would go for the randomization test and cross-check with the hplm (randomization tests tend to have decreased power when a slope effect is prevalent)
This does still not meet the 80% Power and 5% alpha-error convention but is much more informative.

Power in single-case studies is a complex topic.
Here is a study we did on regression analysis that shows some of that complexity:

https://www.frontiersin.org/articles/10.3389/feduc.2022.917944/full

**Reply by gjones1219** [reply]
Thank you for this quick and helpful response! I am very appreciative. However, I have some follow-up questions as I am running this power analysis specifically to detect a potentially small effect using the single case design. My concern is that it does not seem like there are any reasonable permutations of the power analysis that exists within scan that allow for one to detect a small effect. Please see the below example.

set.seed(1)
design <- design(
n = 3,
phase_design = list(A = 29, B = 84),
level = list(A = 0, B = .5),
trend = 0.02,
distribution = "normal",

)

res <- power_test(design, method = c("hplm_slope","tauU_meta", "rand"), n_sim = 100)
res

Output:
Test-Power in percent:

 Method Power Alpha Error Alpha:Beta Correct     p
hplm_slope 7 6 1:15.5 50.5 0.944
tauU_meta 100 100 1:0.0 50.0 1.000
rand 72 4 1:7.0 84.0 0.000

Running this example, the randomization test comes out as seemingly the best, but even with an unreasonably large single case design (i.e. it will be extremely impossible to have a Phase A and a Phase B this long) there is no way to detect a small-to medium size effect using the evaluation convention you suggest (80% power, 5% alpha error), although my understanding is that single case experiments can be useful for exactly this purpose. Can you advise on whether there are other potential power conventions that one can use to evaluate power in a single case design (i.e., cases in which less power and a lower alpha error are okay), or suggest a path forward as I seek to use this power_test function to design an upcoming single case study?

Thank you so much!

Grant

**Reply by jazznbass** [reply]
In your example design the power is low because you defined a trend effect of 0.02 - that is, an increase of 0.02 standarddeviations per(!) measurement. The overall trend effect is 2.26 standard deviations (for 29 + 84 = 113 measurements) which dwarfs the assumed intervention effect of 0.5. tau u cannot handle trend effects.

For hplm, you tested for a slope effect ("hplm_slope") instead of a level effect ("hplm_level")

If you correct both, you get a high power

```.r
set.seed(1)
design <- design(
  n = 3,
  phase_design = list(A = 29, B = 84),
  level = list(A = 0, B = .5),
  trend = 0.2 / 113,
  distribution = "normal",
)

power_test(design, method = c("hplm_level","tauU_meta", "rand"), n_sim = 100)

Test-Power in percent:

     Method Power Alpha Error Alpha:Beta Correct
 hplm_level    97           5      1:0.6      96
  tauU_meta   100          32      1:0.0      84
       rand    99           2      1:0.5      98

```

**Reply by gjones1219** [reply]
Thank you for this response!
